### ISSUE TITLE
Can I see evaluation metrics during training and send them to W&B?

### ISSUE THREAD

**Issue by galtay** [ThilinaRajapakse/simpletransformers#344]
I've been having a great time playing with the library, nice work! 

I was wondering if I'm doing something wrong in this gist? 
https://gist.github.com/galtay/10852bb03b354b2562997973bc29c679

I'm hoping to monitor metrics like "roc_auc_score" during training (and hopefully send them to a weights and biases project).  When I run that code I see the "Running loss" printed out, but not the extra metrics I specified.  Is there a way to log these to the screen or file or wandb? 

I do get the metrics in the `model.eval_model(eval_df, **eval_metrics)` call, but not during training.  Also it would be nice if metrics that take predictions instead of probabilities (like `sklearn.metrics.f1_score`) could be calculated using the user defined per class threshold. 
https://github.com/ThilinaRajapakse/simpletransformers#special-attributes

great package, thanks!

**Comment by ThilinaRajapakse** [comment]
The metrics are logged, not printed. So, you need to set up a logger to see them.

```python
import logging


logging.basicConfig(level=logging.INFO)
transformers_logger = logging.getLogger("transformers")
transformers_logger.setLevel(logging.WARNING)
```
I typically use this.

You can send the metrics to W&B during training by specifying the project name in the args.

```python
train_args = {
    "evaluate_during_training_steps": 1,
    "evaluate_during_training_verbose": True,
    "logging_steps": 1,
    "train_batch_size": 1,
    "eval_batch_size": 1,
    "num_train_epochs": 4,

   "wandb_project": "Your project name",
}
```
You can also configure more wandb options by passing in a dictionary `wandb_kwargs`. These can be any kwargs supported by wandb. For example, to name a run, you can do;

```python
train_args = {
    "evaluate_during_training_steps": 1,
    "evaluate_during_training_verbose": True,
    "logging_steps": 1,
    "train_batch_size": 1,
    "eval_batch_size": 1,
    "num_train_epochs": 4,

   "wandb_project": "Your project name",
   "wandb_kwargs": {"name": "Your run name"}
}
```

You can wrap the `f1_score` or other function in a custom function where the probabilities are converted to 0s or 1s before calling the actual metric function. Then, you can provide your custom function as a `kwarg` to the `eval_model()` call.

**Comment by galtay** [comment]
Thanks for the detailed answer! I like the idea of wrapping the score functions, that's much better than altering the simpletransformers package. 

I gave the logging a try in a colab notebook (https://colab.research.google.com/drive/1e-yJohHyRV_y7k_mQBb64sVdoKPrZwNX?usp=sharing) and am not seeing the custom metrics in the logging or in the weights and biases UI (https://app.wandb.ai/gabrielaltay/simpletransformers_debug?workspace=user-gabrielaltay).  Should I change something? I'm seeing INFO level logging statements (e.g. `INFO:wandb.run_manager:system metrics and metadata threads started`) but nothing about the roc_auc_score, or average_precision _score. 

best, 
-Gabriel

**Comment by ThilinaRajapakse** [comment]
You also need to set `"{evaluate_during_training": True}` in either the `model_args` or `training_args`.

I think that should fix it.

**Comment by galtay** [comment]
Ah! I should have caught that earlier, thanks.  Adding that I now see the default LRAP metrics and the "eval_loss" but not the custom metrics, 

`INFO:simpletransformers.classification.classification_model:{'LRAP': 0.75, 'eval_loss': 0.6888818570545742}`

Is this the correct way to pass custom metrics? 
```
eval_metrics = {
#    "f1": sklearn.metrics.f1_score,
    "roc_auc": sklearn.metrics.roc_auc_score,
    "avg_prc": sklearn.metrics.average_precision_score,
}
model.train_model(
    train_df,
    eval_df=eval_df,
    args=train_args,
    **eval_metrics,
)
```

**Comment by galtay** [comment]
here's the updated colab notebook and wandb site, 
https://colab.research.google.com/drive/1e-yJohHyRV_y7k_mQBb64sVdoKPrZwNX?usp=sharing
https://app.wandb.ai/gabrielaltay/simpletransformers_debug?workspace=user-gabrielaltay

**Comment by galtay** [comment]
I think the problem is here where the kwargs are not passed along 
https://github.com/ThilinaRajapakse/simpletransformers/blob/master/simpletransformers/classification/multi_label_classification_model.py#L168

I can submit a quick PR if you like?

**Comment by galtay** [comment]
https://github.com/ThilinaRajapakse/simpletransformers/pull/351

**Comment by galtay** [comment]
PR is now merged, thanks @ThilinaRajapakse !
